components/cloud/azure/src/azblob.rs

```diff
@@ -498,28 +494,44 @@ impl AzureStorage {
     }
 
     pub fn new(config: Config) -> io::Result<AzureStorage> {
+        let bucket = (*config.bucket.bucket).to_owned();
         // priority: explicit shared key > env Azure AD > env shared key
         if let Some(connection_string) = config.parse_plaintext_account_url() {
-            let bucket = (*config.bucket.bucket).to_owned();
-            let http_client = new_http_client();
-            let container_client = StorageAccountClient::new_connection_string(
-                http_client.clone(),
-                connection_string.as_str(),
-            )
-            .map_err(|e| io::Error::new(io::ErrorKind::InvalidInput, format!("{}", &e)))?
-            .as_storage_client()
-            .as_container_client(bucket);
+            #[cfg(test)]
+            #[cfg(feature = "azurite")]
+            {
```

Comment:
What is this block for? And FYI `cfg(test)` doesn't work for dependencies (if you test this function in another crate the behavior will be different), it's weird to have behavior changing silently like this.

Reply:
For testing the Azure blob with a third party tool Azurite(a tool to mock a local server for testing Azure SDK).

Reply:
I looked again, this feature wasn't even registered in Cargo.toml. I think they are just dead code.

That aside, for this change itself, you should probably just separate it into a new function to keep it clean. One of the confusions I have is, say you are using an "Azurite" specific builder here, what if the `parse_plaintext_account_url` is none before? It falls back to normal implementation, which doesn't make sense to me as an outsider. If it really needs to fallback, then you should add plenty of comments to explain why we can use normal implementation to connect to Azurite, but cannot when `parse_plaintext_account_url` is some.